core-tests/shared/src/test/scala/zio/stm/TMapSpec.scala

```diff
@@ -132,6 +132,26 @@ object TMapSpec extends ZIOBaseSpec {
       }
     ),
     suite("transformations")(
+      testM("toList") {
+        val elems = List("a" -> 1, "b" -> 2)
+        val tx =
+          for {
+            tmap <- TMap.make(elems: _*)
+            list <- tmap.toList
+          } yield list
+
+        assertM(tx.commit)(hasSameElements(elems))
```

Comment:
Don't we want to assert here and below that the two collections are equal, not just that they have the same elements? It doesn't really matter for the map but we definitely want the lists to be the same (i.e. same elements in the same order) and the two maps should be equal as well.

Reply:
@adamgfraser Is it requirement for TMap to preserve order? I thought it's only it's implementation detail.

Reply:
It isn’t a requirement to preserve order since you are correct that it is an implementation detail but for the same reason the definition of equality for maps is that they have the same bindings regardless of order.

Reply:
>  we definitely want the lists to be the same (i.e. same elements in the same order)

If I assert for equivalence here, if `tmap.toList` returns in different order (which it can, since TMap doesn't has no such invariant), the test would fail.

Reply:
You're right. The list won't necessarily be in the same order.